### DISCUSSION TITLE
Netlify Deployment 404

### DISCUSSION THREAD

**Discussion by bitcoinvsalts** [vikejs/vike#1774] [Discussion] [answered]
I am trying to deploy my ssr vike-react site to netlify.

I am running into a 404 after deployment.

Here is my netlify.toml file, any suggestions?

```
[build]
  command = "yarn build"   # Runs the build command specified in package.json
  publish = "dist/client"            # Specifies the directory to publish, Vite outputs to 'dist' by default

[[redirects]]
  from = "/*"
  to = "/index.html"
  status = 200
  force = true
```

<img width="495" alt="image" src="https://github.com/user-attachments/assets/201244fd-3555-4227-a04d-0bdbc853377a">

**Comment by brillout** [comment]
I ain't that familiar with Netlify, but I know that users are deploying Vike to Netlify. Keep us updated, so that we can improve the https://vike.dev/netlify docs.

**Reply by bitcoinvsalts** [reply]
I will as soon as I have the answer.

**Reply by bitcoinvsalts** [reply]
It works if I use this vite.config.ts

```
import react from "@vitejs/plugin-react";
import ssr from "vike/plugin";
import { UserConfig } from "vite";

export default {
  plugins: [
    react(),
    ssr({
      prerender: true,
      includeAssetsImportedByServer: true,
    }),
  ],
  optimizeDeps: { include: ["react/jsx-runtime"] },
} satisfies UserConfig;
```

and netlify.toml:

```
[build]
  publish = "dist/client/"
```

**Reply by brillout** [reply]
Thank you!  If you're up for it, a small example repository would be great :eyes:

**Reply by bitcoinvsalts** [reply]
I still have an issue with those pages: https://directbooking.netlify.app/star-wars/4

I am not sure why yet.

**Reply by brillout** [reply]
Have a look at https://vike.dev/onBeforePrerenderStart.

**Comment by bitcoinvsalts** [comment]
Hello @brillout ,

Any suggestion on how to deploy a vike + express server like this one https://github.com/bitcoinvsalts/vavike ?

I tried with Netlify and Vercel without success.

**Reply by brillout** [reply]
https://vike.dev/express  
https://brillout.github.io/rules#try

**Reply by bitcoinvsalts** [reply]
Trust me I tried many ways and already spent more than 6 hours without success. I cannot find a find to deploy it on vercel, neither netlify. 

I wonder what other deployment options people are using?

**Reply by brillout** [reply]
It's trivial to deploy a vanilla Vike + Express.js app. I guess you're refering to [Vike + vike-node + Express.js](https://github.com/vikejs/vike-node/issues/4).

**Reply by bitcoinvsalts** [reply]
No, I am trying every possible way. Could you provide your trivial solution for a vanilla Vike + Express.js app? @brillout

**Reply by brillout** [reply]
I assume you have seen https://vike.dev/vercel? What's the issue you're having?

**Reply by bitcoinvsalts** [reply, marked as answer]
I found the solution to my deployment issue!

If I use ```prerender: true``` in the vite config file like this, it wont process the destination API correctly while using auth

```
plugins: [
    vike({
      prerender: true,
    }),
    //mdx(),
    react()
  ],
```

It works fine if I set **prerender** to **false**. 

Big thanks to @brillout for your help.
